src/main.rs

```diff
@@ -157,3 +244,47 @@ fn parse_line(line: &str) -> anyhow::Result<Option<DVec3>> {
         _ => Ok(None),
     }
 }
+
+fn histogram(
+    positions: &Vec<DVec3>,
+    r_variable: fn(&DVec3) -> f64,
+    (avg, std_dev): (DVec3, DVec3),
+) -> (Vec<Vec<DVec3>>, Vec<(f64, f64)>) {
+    let mut position_set: Vec<DVec3> = positions.clone();
+
+    let cutoff: i32 = 3; // measured in standard deviations
+    let div: i32 = 6;
+    let mut range = (-(cutoff * div)..(cutoff * div))
+        .into_iter()
+        .map(|i| (i as f64) / (div as f64) * r_variable(&std_dev) + r_variable(&avg))
+        .peekable();
+    position_set.sort_by(|a, b| {
+        r_variable(a)
+            .partial_cmp(&r_variable(b))
+            .expect("Histogram: Incomparable values")
```

Comment:
Return an error instead of panicking here. You can use `anyhow::anyhow!` to create an error with an arbitrary message.